### PR TITLE
Added ConstructClause to Statement struct

### PR DESCRIPTION
**Pull request by palimarrao**
I want to make sure that the addition I am making to the `Statement` struct looks good.

### CODE REVIEW

File: bql/semantic/semantic.go [hunk 1]
```diff
@@ -75,6 +79,8 @@ type Statement struct {
 	data                      []*triple.Triple
 	pattern                   []*GraphClause
 	workingClause             *GraphClause
+	constructClauses          []*ConstructClause
```
Comment (xllora) [inline bql/semantic/semantic.go]: Could you elaborate why you need a new working clause? The where clause of a construct is the  same as the one on a regular select. I was wondering if it is just cleaner to just add the actual construct information containing the info to create the new triples. Or am I just mislead by the name?

File: bql/semantic/semantic.go [hunk 2]
```diff
@@ -122,6 +128,33 @@ type GraphClause struct {
 	OTemporal        bool
 }
 
+// ConstructClause represents a singular clause within a construct statement.
+type ConstructClause struct {
```
Comment (xllora) [inline bql/semantic/semantic.go]: Despite the name this is just the info on the construct right? Just want to make sure I got it right :)

Comment (palimarrao) [inline bql/semantic/semantic.go]: Yep, it is just the information required to insert a new triple into the graph. For instance (a contrived example):

`construct {?s ?p ?o; ?p1 ?o1} into ?a from ?b where {?s ?p ?o. ?s ?p1 ?o2}`

will lead to the creation of the following instance of `constructClauses`:

```
[]*ConstructClauses{
  {
    SBinding: ?s,
    PBinding: ?p,
    OBinding: ?o,
    ReificationClauses: []*ReificationClause{
      {
         PBinding: ?p1
         OBinding: ?o1
      },
   },
}

```
I guess the name is confusing. Do you have any suggestions?

File: bql/semantic/semantic.go (hunk 1 above)
Comment (palimarrao) [inline bql/semantic/semantic.go]: So `constructClauses` captures only the declarations within a `CONSTRUCT` statement that specifies how the triples must be built. For instance, consider the following:

`construct {?s "new_predicate"@[] ?o} into ?a from ?b where {?s "old_predicate"@[] ?o}`

`constructClauses` simply captures the `?s "new_predicate"@[] ?o` part. `workingClauses` will capture the `?s "old_predicate"@[] ?o` as before. `constructClauses` will capture multiple such declarations (separated by the `.` operator) and `constructClause` also has support for multiple reification clauses. So in a way, `constructClause` is similar to `data` within the `Statement` struct (except it has to support literals, bindings and reification). Does this seem like the right thing to do?

**Pull request merged**